File: PlayTen-Client/playten/src/app/models/tournament.ts
import { Place } from "./place";
import { Participant } from "./participant";

export class Tournament {
 public tournament!: TournamentInfo;
 public isUserTournamentOwner!: boolean;
 public isUserParticipant!: boolean;
 public isUserApprovedParticipant!: boolean;
 public isUserUndeterminedParticipant!: boolean;
 public isUserRejectedParticipant!: boolean;
 public isTournamentFinished!: boolean;
}

export class TournamentInfo {
 public tournamentId?: number;
 public amountOfRounds?: number;
 public name!: string;
 public dateStart!: Date;
 public dateEnd!: Date;
 public description!: string;
 public placeId!: number;
 public numberOfParticipants!: number;
 public ownerId!: string;
 public place?: Place;
 public participants?: Participant[];
 public price?: string;
 public hasStarted?: boolean;
 public finished?: boolean;
}

File: PlayTen/PlayTen.DAL/Entities/Tournament.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace PlayTen.DAL.Entities
{
    public class Tournament
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public DateTime DateStart { get; set; }
        public DateTime DateEnd { get; set; }
        public bool HasStarted { get; set; }
        public bool Finished { get; set; }
        public int AmountOfRounds { get; set; }
        public int NumberOfParticipants { get; set; }
        public string Description { get; set; }
        public string Price { get; set; }
        public int PlaceId { get; set; }
        public string OwnerId { get; set; }
        public ICollection<Participant> Participants { get; set; }
        public Place Place { get; set; }
        public User User { get; set; }
    }
}
